### DISCUSSION TITLE
Why is my sidebar not sticky?

### DISCUSSION THREAD

**Discussion by martinszeltins** [tailwindlabs/tailwindcss#5540] [Help]
I have created a simple 2 column grid where I wanted the sidebar to stick to the top as I start scrolling down but it is not working. What am I doing wrong here?

Here is my Tailwind Play snippet - https://play.tailwindcss.com/zkr9N7Jwjl

**Comment by cairin** [comment]
If you inspect the `div` surrounding your sidebar you'll see its height spans the height of the parent.

Simply giving it a height makes it sticky within the grid column: https://play.tailwindcss.com/F35RwOnh8K

**Reply by martinszeltins** [reply]
> If you inspect the `div` surrounding your sidebar you'll see its height spans the height of the parent.
> 
> Simply giving it a height makes it sticky within the grid column: https://play.tailwindcss.com/F35RwOnh8K

Setting a fixed height on things is not the best idea. Imagine if the sidebar will have more content that fits into 21px

**Reply by cairin** [reply]
Sure, but then you should be using flex, not grid.

**Reply by martinszeltins** [reply]
> Sure, but then you should be using flex, not grid.

Why would I use flexbox for layouts? CSS Grid is way more flexible (pun intended) than flexbox.

**Reply by cairin** [reply]
> Why would I use flexbox for layouts?

That's a good question. I would say it depends entirely on your use-case, and what the content of your layout will be. 
[Here's a good article on the relationship between the two](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Relationship_of_Grid_Layout#grid_and_flexbox). Specifically this section:

> ### Content out or layout in?
> In addition to the one-dimensional versus two-dimensional distinction, there is another way to decide if you should use flexbox or grid for a layout. Flexbox works from the content out. An ideal use case for flexbox is when you have a set of items and want to space them out evenly in a container. You let the size of the content decide how much individual space each item takes up. If the items wrap onto a new line, they will work out their spacing based on their size and the available space on that line.
> 
> Grid works from the layout in. When you use CSS Grid Layout you create a layout and then you place items into it, or you allow the auto-placement rules to place the items into the grid cells according to that strict grid. It is possible to create tracks that respond to the size of the content, however, they will also change the entire track.
> 
> If you are using flexbox and find yourself disabling some of the flexibility, you probably need to use CSS Grid Layout. An example would be if you are setting a percentage width on a flex item to make it line up with other items in a row above. In that case, a grid is likely to be a better choice.

I think it's also important to note that there are many css [layout](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Introduction) tools. A layout can be scaffolded using either grids and flexboxes, both or neither.

> CSS Grid is way more flexible (pun intended) than flexbox.

It's a lot more nuanced than that. As an analogy one could say that an adjustable wrench is more flexible than a screwdriver. You **can** use the wrench to loosen a screw, whether you **should** is debatable.
